src/server/routes/plywood/plywood.ts

```diff
@@ -17,16 +17,44 @@
 
 import { Timezone } from "chronoshift";
 import { Request, Response, Router } from "express";
-import { Dataset, Expression } from "plywood";
+import { IncomingHttpHeaders } from "http";
+import { Dataset, Expression, RefExpression } from "plywood";
+import { isFunction } from "util";
+import { DynamicSubsetFormula } from "../../../common/models/dynamic-subset-formula/dynamic-subset-formula";
+import { isNil } from "../../../common/utils/general/general";
 import { checkAccess } from "../../utils/datacube-guard/datacube-guard";
 import { SettingsGetter } from "../../utils/settings-manager/settings-manager";
 
+function filterMain(expression: Expression, filter: Expression): Expression {
```

Comment:
I would move filterMain and applySubset somewhere. Route is not the place for this kind of logic.

Reply:
Of course!